File: solutions/1512. Number of Good Pairs/1512_2.py
# Time complexity: O(n)
# Space complexity: O(n)
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        counts = defaultdict(int)
        ans = 0

        for num in nums:
            ans += counts[num]
            counts[num] += 1

        return ans    

File: solutions/1512. Number of Good Pairs/1512_2.cs
// Time complexity: O(n)
// Space complexity: O(n)
public class Solution {
    public int NumIdenticalPairs(int[] nums) {
        Dictionary<int, int> counts = new Dictionary<int, int>();
        int ans = 0;
        
        foreach (int num in nums) {
            if (counts.ContainsKey(num)) {
                ans += counts[num];
                counts[num]++;
            }
            else {
                counts[num] = 1;
            }
        }
        
        return ans;
    }
}